File: OCH_Tao/ABC/A/308.py
S = list(map(int,input().split()))
flag = True
for i in range(7):
  if S[i]>S[i+1]:
    flag=False
    break
if min(S)<100 or max(S)>675:
  flag=False
if any([i%25!=0 for i in S]):
  flag=False
if flag:
  print("Yes")
else:
  print("No")

File: OCH_Tao/ABC/A/308.js
const S = (require("fs").readFileSync("/dev/stdin", "utf8").trim().split("\n"))[0].split(" ").map(x => parseInt(x, 10));
let flag = true;
for (let i = 0; i < 7; i++) {
  if (S[i] > S[i + 1]) {
    flag = false;
    break;
  }
}
if (Math.min(...S) < 100 || Math.max(...S) > 675) {
  flag = false;
}
if (S.some(x => x % 25 !== 0)) {
  flag = false;
}
if (flag) {
  console.log("Yes");
} else {
  console.log("No");
}